go/vt/vtgate/planbuilder/operators/queryprojection.go

```diff
@@ -112,14 +111,11 @@ type (
 
 func (aggr Aggr) NeedWeightString(ctx *plancontext.PlanningContext) bool {
 	switch aggr.OpCode {
-	case opcode.AggregateCountDistinct, opcode.AggregateSumDistinct:
+	case opcode.AggregateCountDistinct, opcode.AggregateSumDistinct, opcode.AggregateMin, opcode.AggregateMax:
 		return ctx.SemTable.NeedsWeightString(aggr.Func.GetArg())
-	case opcode.AggregateMin, opcode.AggregateMax, opcode.AggregateGroupConcat:
-		// currently this returns false, as aggregation engine primitive does not support the usage of weight_string
-		// for comparison. If Min/Max column is non-comparable then it will fail at runtime.
```

Comment:
should this be kept as-is for now?

Reply:
nah, we should fix the engine primitive instead. let's just fail correctly while running the query if it turns out we really need the weightstring instead